electron/main/llm/llmConfig.ts

```diff
@@ -33,3 +32,77 @@ export function validateAIModelConfig(
 
   return null;
 }
+
+export async function addOrUpdateLLMSchemaInStore(
+  store: Store<StoreSchema>,
+  modelConfig: LLMConfig
+): Promise<void> {
+  const existingModelsInStore = await store.get(StoreKeys.LLMs);
+  console.log("existingModels: ", existingModelsInStore);
+  const isNotValid = validateAIModelConfig(modelConfig);
+  if (isNotValid) {
+    throw new Error(isNotValid);
+  }
+
+  const foundModel = existingModelsInStore.find(
+    (model) => model.modelName === modelConfig.modelName
+  );
+
+  console.log("foundModel: ", foundModel);
+
+  if (foundModel) {
+    console.log("updating model");
+    const updatedModels = existingModelsInStore.map((model) =>
+      model.modelName === modelConfig.modelName ? modelConfig : model
+    );
+    store.set(StoreKeys.LLMs, updatedModels);
+  } else {
+    console.log("adding model");
+    const updatedModels = [...existingModelsInStore, modelConfig];
+    store.set(StoreKeys.LLMs, updatedModels);
+  }
+}
+
+export async function removeLLM(
+  store: Store<StoreSchema>,
+  ollamaService: OllamaService,
+  modelName: string
+): Promise<void> {
+  const existingModels = (store.get(StoreKeys.LLMs) as LLMConfig[]) || [];
+
+  const foundModel = await getLLMConfig(store, ollamaService, modelName);
+
+  if (!foundModel) {
+    return;
+  }
+
+  const updatedModels = existingModels.filter(
+    (model) => model.modelName !== modelName
+  );
+  store.set(StoreKeys.LLMs, updatedModels);
+
+  ollamaService.deleteModel(modelName);
+}
+
+export async function getAllLLMConfigs(
+  store: Store<StoreSchema>,
+  ollamaSession: OllamaService
+): Promise<LLMConfig[]> {
+  const llmConfigsFromStore = store.get(StoreKeys.LLMs);
+  const ollamaLLMConfigs = await ollamaSession?.getAvailableModels();
+
+  return [...llmConfigsFromStore, ...ollamaLLMConfigs];
```

Comment:
If I understand this correctly, are the store LLM configs for remote models, while the Ollama configs are for local models?

Reply:
Yes we will now just rely on and query ollama's store for local models and remote models will be managed by us